ad9656: add reference project for the ad9656 eval board

hotoleanudan opened the pull request:
Added new reference project for ad9656 eval board coupled with the zcu102 carrier.

Signed-off-by: Dan Hotoleanu <dan.hotoleanu@analog.com>

amiclaus commented on projects/ad9656_fmc/src/app/ad9656_fmc.c:
```diff
@@ -0,0 +1,530 @@
+	
+	// configure 9508 to pass the 125MHz input clock unmodified, so divider = 1
+	// no phase offset
+	ad9508_spi_write(dev, 0x1B, 0x00); // divide ratio[7:0]
```
Would be nice to use macro defines for the Reg addresses.

amiclaus replied:
This statement is valid for the entire commit.

hotoleanudan replied:
Changed hardcoded register numbers to defined constants.

amiclaus commented on projects/ad9656_fmc/src/app/app_config.h:
```diff
@@ -0,0 +1,43 @@
+
+#ifndef APP_CONFIG_H_
+#define APP_CONFIG_H_
+
```
This file will be populated in the future when support for new platforms / features will be added?
Otherwise it is redundant.

hotoleanudan replied:
I discussed with Adrian Costina and he told me that in the near future there will be no support for new platforms / features. But still, I would keep this file in order the preserve the file structure.

amiclaus commented on projects/ad9656_fmc/src/devices/adi_hal/parameters.h:
```diff
@@ -0,0 +1,55 @@
+#ifndef _PARAMETERS_H_
+#define _PARAMETERS_H_
+
+#include "app_config.h"
```
You don't need to include `app_config.h`

hotoleanudan replied:
I would keep this file in order the preserve the file structure of the folder.

hotoleanudan replied:
Removed the include.

amiclaus commented:
Fix please code style issues detected by the travis build.

amiclaus commented on projects/ad9656_fmc/src/app/ad9656_fmc.c:
```diff
@@ -0,0 +1,530 @@
+#define DMA_BUFFER		 0
+
+struct ad9508_init_param {
+	/* SPI */
```
Just an opinion. I would separate the device driver implementations(ad9553, ad9508) into separate files.
Even though the implementations are minimal, it will be easier to expand them afterwards and add to the `drivers` folder.

hotoleanudan replied:
Moved the driver implementation for the ad9553 and ad9508 devices into separate files in ad9656_fmc/src/app folder.

hotoleanudan commented:
Can you please give me a feedback on the last update that I made to this project? Or if there are no more comments on it can I merge it on master branch?

hotoleanudan commented:
Changelog v2:
- moved the driver code for ad9508 device from ad9656_fmc/src/app/ad9656_fmc.c to ad9656_fmc/src/app/ad9508.c and ad9656_fmc/src/app/ad9508.h
- moved the driver code for ad9553 device from ad9656_fmc/src/app/ad9656_fmc.c to ad9656_fmc/src/app/ad9553.c and ad9656_fmc/src/app/ad9553.h
- used defined constants for addresses of registers instead of their adresses (names instead of numbers)
- deleted the "include app_config.h" instruction from the parameters.h

hotoleanudan commented:
Changelog v3:
- Modifed the project files such that it will comply with the AStyle

hotoleanudan commented:
Changelog v4:

- modified the text indentation for Makefile and src.mk files

adrimbarean commented on projects/ad9656_fmc/src/app/ad9508.c:
```diff
@@ -0,0 +1,170 @@
+ * @param reg_data - The value read from the internal register
+ * @return SUCCESS if the value was successfully read, FAILURE otherwise
+ */
+int32_t ad9508_spi_read(struct ad9508_dev *dev,
```
I would rename these `ad9508_reg_read` and `ad9508_reg_write` to be consistent with other drivers. Also, I think these files can be added as drivers in the `drivers` folder and included from there.

hotoleanudan replied:
Done in v6

adrimbarean commented on projects/ad9656_fmc/src/app/ad9508.c:
```diff
@@ -0,0 +1,170 @@
+ * @param init_param - Values for the working parameters of ad9508
+ * @return SUCCESS if device is ready for use, FAILURE otherwise
+ */
+int ad9508_setup(struct ad9508_dev **device,
```
nitpick: 
```suggestion
int32_t ad9508_setup(struct ad9508_dev **device,
```

hotoleanudan replied:
Done in v6

adrimbarean commented on projects/ad9656_fmc/src/app/ad9656_fmc.c:
```diff
@@ -0,0 +1,265 @@
+		printf("The ad9553 chip successfully configured\n");
+
+	// ADC
+	ad9656_setup(&ad9656_device, &ad9656_param);
```
I would check for this return value as well.

hotoleanudan replied:
Done in v6

hotoleanudan commented:
Changelog v5:

- Squased the commits on the branch on the first commit and modified the resulting commit's comment.

hotoleanudan commented:
Changelog v6:

- Changed function names ad9508_spi_read, ad9508_spi_write, ad9553_spi_read and ad9553_spi_write to ad9508_reg_read, ad9508_reg_write, ad9553_reg_read and ad9553_reg_write. Also changed all their occurrences.
- Changed the return type of ad9508_setup and ad9553_setup from int to int32_t.
- Checked the return value of the ad9656_setup function call in ad9656_fmc.c file.

mirceacaprioru commented on projects/ad9656_fmc/src/app/ad9508.c:
```diff
@@ -0,0 +1,191 @@
+			uint8_t *reg_data)
+{
+	uint8_t buf[3];
+
```
nitpick: remove extra line

hotoleanudan replied:
Done in v7

mirceacaprioru commented on projects/ad9656_fmc/src/app/ad9508.c:
```diff
@@ -0,0 +1,191 @@
+
+	int32_t ret;
+
+	// the MSB of byte 0 indicates a r/w operation, following 7 bits are the
```
multi line comment
```
/*
 *
 */

hotoleanudan replied:
Done in v7

mirceacaprioru commented on projects/ad9656_fmc/src/app/ad9508.c:
```diff
@@ -0,0 +1,191 @@
+ ********************************************************************************
+ * Copyright 2020(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
```
looking better here aren't these the driver files ? You don't need to copy them here again. Just use the ones from drivers folder.

hotoleanudan replied:
These drivers are new, they are written by me.

mirceacaprioru commented on projects/ad9656_fmc/Makefile:
```diff
@@ -0,0 +1,6 @@
+TARGET := ad9656_fmc
```
you have a typo in the commit message: "recevied" -> "received"

hotoleanudan replied:
Changed in v7

mirceacaprioru commented on projects/ad9656_fmc/src/app/ad9553.c:
```diff
@@ -0,0 +1,244 @@
+/***************************************************************************//**
+ * @file ad9553.c
+ * @brief Implementation of AD9553 Driver.
+ * @author DHotolea (dan.hotoleanu@analog.com)
```
It seems that you also added some new drivers with this PR. Separate the drivers in different commits and then just add a commit with the project.

hotoleanudan replied:
I discussed with Dragos Bogdan and we will leave it like it is until there will be another project that will require one of the two drivers. In that situation the setup functions will be modified to be more generic and the drivers will be included in the drivers folder.

mirceacaprioru replied:
Ack.

hotoleanudan commented:
Changelog v7:

- corrected commenting style and an extra line in ad9508.c.

mirceacaprioru commented:
Please do a rebase before merging this in order to fix the CI.

hotoleanudan commented:
Changelog v8:

- done a rebase.
